apps/flame-defi/app/earn/modules/market-list/sections/content-section.tsx

```diff
@@ -0,0 +1,82 @@
+import { Pagination, Skeleton, StatusCard } from "@repo/ui/components";
+import { getPlaceholderData, MarketListTable } from "earn/components/market";
+import { SearchInput } from "earn/components/search-input";
+import { Market, OrderDirection } from "earn/generated/gql/graphql";
+import { PAGE_SIZE } from "earn/modules/market-list/hooks/use-fetch-markets";
+import { usePageContext } from "earn/modules/market-list/hooks/use-page-context";
+import { useMemo } from "react";
+
+export const ContentSection = () => {
+  const {
+    currentPage,
+    setCurrentPage,
+    ordering,
+    sorting,
+    setSorting,
+    search,
+    setSearch,
+    query: { data, isPending, isRefetching },
+    status,
+  } = usePageContext();
+
+  const totalPages = useMemo(() => {
+    return Math.ceil((data?.markets?.pageInfo?.countTotal ?? 0) / PAGE_SIZE);
```

Comment:
Are these types from generated? Not loving the nested optional properties, but makes sense if it's from gql